File: week8.cpp
#include <iostream>
using namespace std;
// Base class
class Square{
protected:
    double side;
public:
    // Constructor
    Square(double s=0){
        side=s;
    }
    // Destructor
    ~Square(){
    }
    double getPeri(){
        return 4*side;
    }
    double getArea(){
        return side*side;
    }
};

// Derived class
class Cube:public Square{
public:
    // Constructor
    Cube(double s=0) : Square(s){
    }
    // Destructor
    ~Cube(){
    }
    double getArea(){
        return 6*side*side;
    }

    double getVolume(){
        return side*side*side;
    }
};

int main() {
    Square sq(15);
    cout <<"Square Perimeter:"<<sq.getPeri()<<endl;
    cout <<"Square Area:"<< sq.getArea()<<endl;

    Cube cb(15);
    cout << "Cube Surface Area:"<<cb.getArea()<<endl;
    cout << "Cube Volume:"<<cb.getVolume()<<endl;

    return 0;
}

File: week8.py
# Base class
class Square:
    def __init__(self, side=0):
        self.side = side

    def getPeri(self):
        return 4 * self.side

    def getArea(self):
        return self.side * self.side
# Derived class
class Cube(Square):
    def __init__(self, side=0):
        super().__init__(side)

    def getArea(self):
        return 6 * self.side * self.side

    def getVolume(self):
        return self.side * self.side * self.side
# Main program
sq = Square(15)
print("Square Perimeter:", sq.getPeri())
print("Square Area:", sq.getArea())

cb = Cube(15)
print("Cube Surface Area:", cb.getArea())
print("Cube Volume:", cb.getVolume())
